vm/compiler/src/ledger/mod.rs

```diff
@@ -594,11 +591,38 @@ impl<N: Network, B: BlockStorage<N>, P: ProgramStorage<N>> Ledger<N, B, P> {
         Ok(())
     }
 
+    /// Adds a given address to the validator set.
+    pub fn add_validator(&mut self, address: Address<N>) -> Result<()> {
+        if self.validators().contains_key(&address) {
+            bail!("'{address}' is already in the validator set.")
+        }
+
+        // Insert the address into the validator set.
+        self.validators.insert(address, ());
+        Ok(())
+    }
+
+    /// Removes a given address from the validator set.
+    pub fn remove_validator(&mut self, address: Address<N>) -> Result<()> {
+        if !self.validators().contains_key(&address) {
```

Comment:
ditto: `remove` indicates whether the key had existed, so a double lookup is not necessary

Reply:
Same answer as above, however I guess using the `Option<V>` output of the `remove` call could be better than the double lookup.

Reply:
it would look like this:
```
pub fn remove_validator(&mut self, address: Address<N>) -> Result<()> {
    if self.validators.remove(&address).is_none() {
        bail!("'{address}' is not in the validator set.")
    } else {
        Ok(())
    }
}
```